compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -553,9 +711,139 @@ impl SirFuncCx<'tcx> {
         }
     }
 
-    fn lower_ref(&self, place: &mir::Place<'_>) -> ykpack::Rvalue {
-        let sir_place = self.lower_place(place);
-        ykpack::Rvalue::Ref(sir_place)
+    fn lower_ref<Bx: BuilderMethods<'a, 'tcx>>(
+        &mut self,
+        bx: &Bx,
+        bb: ykpack::BasicBlockIndex,
+        dest_ty: Ty<'tcx>,
+        place: &mir::Place<'tcx>,
+    ) -> ykpack::IPlace {
+        let ty = self.lower_ty_and_layout(bx, &self.mono_layout_of(bx, dest_ty));
+        let dest_ip = self.new_sir_local(ty);
+        let src_ip = self.lower_place(bx, bb, place);
+        let mkref = ykpack::Statement::MkRef(dest_ip.clone(), src_ip.clone());
+        if let Some(src_local) = src_ip.local() {
+            self.notify_referenced(src_local);
+        }
+        self.push_stmt(bb, mkref);
+        dest_ip
+    }
+
+    fn lower_ty_and_layout<'a, Bx: BuilderMethods<'a, 'tcx>>(
+        &mut self,
+        bx: &Bx,
+        ty_layout: &TyAndLayout<'tcx>,
+    ) -> ykpack::TypeId {
+        let sir_ty = match ty_layout.ty.kind() {
+            ty::Int(si) => self.lower_signed_int_ty(*si),
+            ty::Uint(ui) => self.lower_unsigned_int_ty(*ui),
+            ty::Adt(adt_def, ..) => self.lower_adt_ty(bx, adt_def, &ty_layout),
+            ty::Array(elem_ty, len) => {
+                let align = usize::try_from(ty_layout.layout.align.abi.bytes()).unwrap();
+                let size = usize::try_from(ty_layout.layout.size.bytes()).unwrap();
+                ykpack::Ty::Array {
+                    elem_ty: self.lower_ty_and_layout(bx, &self.mono_layout_of(bx, elem_ty)),
+                    len: usize::try_from(len.eval_usize(self.tcx, ty::ParamEnv::reveal_all()))
+                        .unwrap(),
+                    byte_size: size, // FIXME use SizeAndAlign
```

Comment:
I also spotted this one reviewing the FIXMEs. This should be fixed too.

Reply:
Fixed in 0fe6394ae110bdd0e9bfa3748ed8a8a68b6dd320 -- there's a yk change for that too.